.github/workflows/ci.yml

```diff
@@ -14,21 +12,26 @@ env:
 jobs:
   tests:
     name: Run tests (${{ matrix.image }})
+    # Run the job ONLY IF:
+    # 1. It is a pull_request event OR
+    # 2. It is a push event AND the repository is not a fork
+    # The push event for a non-fork PR branch will typically be a duplicate, so we exclude it.
+    if: github.event_name == 'pull_request' || github.ref != format('refs/heads/{0}', github.event.pull_request.head.ref)
 
```

Comment:
I'm not sure if I understand the condition properly. I found the following which seems to make more sense to me: https://github.com/orgs/community/discussions/57827#discussioncomment-6579237